test/ui/BreadcrumbValuesListTest.ts

```diff
@@ -0,0 +1,113 @@
+import { $$ } from '../Test';
+import { FakeResults } from '../Fake';
+import { IIndexFieldValue } from '../../src/rest/FieldValue';
+
+export function BreadcrumbValueListTest() {
+  describe('BreadcrumbValuesListTest', () => {
+    let facet: Facet;
+    let facetValues: FacetValue[];
+
+    beforeEach(() => {
+      facet = Mock.mock(Facet);
+      facet.options = {};
+      facet.options.numberOfValuesInBreadcrumb = 5;
+      facet.getValueCaption = jasmine.createSpy('getValueCaption').and.callFake((facetValue: IIndexFieldValue) => facetValue.value);
+      facetValues = _.map(FakeResults.createFakeFieldValues('foo', 3), fieldValue => {
+        const created = FacetValue.createFromFieldValue(fieldValue);
+        created.selected = true;
+        return created;
+      });
+    });
+
+    it('should support rendering as a simple string and not an HTML element', () => {
+      facet.options.title = 'My facet title';
+      const breadcrumbList = new BreadcrumbValueList(facet, facetValues, BreadcrumbValueElement);
+      const builtAsString = breadcrumbList.buildAsString();
+      expect(_.isString(builtAsString)).toBe(true);
+      expect(builtAsString).toContain('My facet title');
+      expect(builtAsString).toContain('foo0');
+      expect(builtAsString).toContain('foo1');
+      expect(builtAsString).toContain('foo2');
+    });
+
+    it('should output the title with the facet title option', () => {
+      facet.options.title = 'My facet title';
+      const breadcrumbList = new BreadcrumbValueList(facet, facetValues, BreadcrumbValueElement);
+      const built = breadcrumbList.build();
+      const title = $$(built).find('.coveo-facet-breadcrumb-title');
+      expect($$(title).text()).toContain('My facet title');
+    });
+
+    it('should display a clickable label for each selected values', () => {
+      const breadcrumbList = new BreadcrumbValueList(facet, facetValues, BreadcrumbValueElement);
+      const built = breadcrumbList.build();
+      const values = $$(built).findAll('.coveo-facet-breadcrumb-value');
+      expect(values.length).toBe(3);
+    });
+
+    it('should display a caption for each selected values', () => {
+      const breadcrumbList = new BreadcrumbValueList(facet, facetValues, BreadcrumbValueElement);
+      const built = breadcrumbList.build();
+      const captions = $$(built).findAll('.coveo-facet-breadcrumb-caption');
+      expect($$(captions[0]).text()).toContain('foo0');
+      expect($$(captions[1]).text()).toContain('foo1');
+      expect($$(captions[2]).text()).toContain('foo2');
+    });
+
+    it('should display a clear button for each selected values', () => {
+      const breadcrumbList = new BreadcrumbValueList(facet, facetValues, BreadcrumbValueElement);
+      const built = breadcrumbList.build();
+      const clear = $$(built).findAll('.coveo-facet-breadcrumb-clear');
+      expect(clear.length).toBe(3);
+    });
+
+    it('should display excluded values differently', () => {
```

Comment:
The tests `should display excluded values differently`, `should add the excluded and selected values together when it exceeds the number of values in breadcrumb` and `should show hidden values on clicking show more` seem like they belong together in a describe block with a beforeEach that sets `.excluded` and `.selected`.